specs/proof_of_capacity.md

```diff
@@ -21,4 +21,27 @@ This shows that today, the validity of the capacity registered is only based on
 
 We need to come up with some algorithm that provide some real proof of capacity. We also need to be able to limit the right to register capacity to only people that are verified and accepted as farmer.
 
-We also need to come up with a way to register a farm and authorizing certain address to manage this farm (as in add themselves to that farm, or be linked to it since creation, even though I'm fairly sure we also might need to add more addresses after creation).
\ No newline at end of file
+We also need to come up with a way to register a farm and authorizing certain address to manage this farm (as in add themselves to that farm, or be linked to it since creation, even though I'm fairly sure we also might need to add more addresses after creation).
+
+
+## Possible solutions
+
+We need to find a solution to measure the capacity of 4 units:
+
+- CPU
+- Memory
+- Hard drive disk storage
+- Solid state drive storage
+
+### CPU and Memory
+For CPU and memory, we could use some kind of computational puzzle that are CPU or memory bound to ensure that the capacity a node reports can indeed be verified by solving one or more of these puzzles.  
+For this to work the difficulty of a puzzle need to be tunable and link with a fix minimum amount of CPU power or amount of memory.
+
+On problem with this approach is that once the node will be used to run workloads, both the CPU and memory will be used and not available to solve puzzle anymore. So we don't have a solution that allow us to re-check the capacity of a node over time while it is being used.
```

Comment:
I don't think this is an issue, as we should only pay for what is used, once it is used. And whatever is using should serve as proof of its own. The thing running on it, or storing the stuff, should be able to proof it for itself in a different but still easy way. The storage use-case could easily request a certain range of bytes randomly at points, as to give an example.

Reply:
>as we should only pay for what is used, once it is used

That's actually not how things works. There are 2 ways of getting token.
1. when you as a farmer provide capacity to the grid
2. when your capacity is used

This document is focusing on the first part, where we need some proof of the capacity that the farmer gives to the grid.

Also I'm not sure your example for storage stands, cause once the capacity is used, it is reserve and private to someone, so in the case of storage, the network has no way to know kind of data is stored and thus no way to validate that the farmer is still able to provide it.